### PR TITLE
Complete re-write of two Xtend file to JAVA ( using modern JAVA

### PR DESCRIPTION
**Pull request by samiura**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .frontmatter/database/mediaDb.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (bjandras) [inline .frontmatter/database/mediaDb.json:1]: What is this file?

File: frontmatter.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (bjandras) [inline frontmatter.json:1]: What is this file?

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import org.eclipse.xtext.xbase.lib.ListExtensions;
+
+public class MessageExtensions {
+  private static String prependCommaIfNotEmpty(String string) {
+    return  string.isEmpty() ? string : ", " + string;
+  }
+
+  public static String prototype(Message msg) {
+    List<Field> sortedFields = msg.getFields().stream()
+			.sorted(Comparator.comparingInt(Field::getId))
```
Comment (bjandras) [inline renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java:29]: Tabs.

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			.collect(Collectors.toList());
+
+    String finalString = sortedFields.stream()
+      .map( it -> it.getType().isIsShortString() ? "const char " + it.getName() + "[" +it.getType().getSize() + "]" + FieldExtensions.arraySuffix(it) : "const " + FieldTypeExtensions.parsedCType(it.getType()) + " " + it.getName() +  FieldExtensions.arraySuffix(it) )
```
Comment (bjandras) [inline renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java:33]: Extra spacing.

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    return finalString;
+  }
+
+  public static String commaPrototype(Message msg) {
+    return MessageExtensions.prependCommaIfNotEmpty(MessageExtensions.prototype(msg));
+  } 
+
+  public static String callPrototype(Message msg) {
+    String finalString = msg.getFields().stream()
+			.sorted(Comparator.comparingInt(Field::getId))
```
Comment (bjandras) [inline renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java:44]: Tabs.

**Review by bjandras — CHANGES_REQUESTED** [review]

**Comment by bjandras** [comment]
So Java has no extension methods concept. Have you considered how that fact will reflect on the remaining code?

File: .frontmatter/database/mediaDb.json (hunk 1 above)
Comment (samiura) [reply .frontmatter/database/mediaDb.json]: removed, came from my vscode extension!

File: frontmatter.json (hunk 2 above)
Comment (samiura) [reply frontmatter.json]: same

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java (hunk 3 above)
Comment (samiura) [reply renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java]: removed

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java (hunk 4 above)
Comment (samiura) [reply renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java]: removed

File: renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java (hunk 5 above)
Comment (samiura) [reply renderc/io.opentelemetry.render/src/io/opentelemetry/render/extensions/MessageExtensions.java]: removed

File: .frontmatter/database/mediaDb.json (hunk 1 above)
Comment (bjandras) [reply .frontmatter/database/mediaDb.json]: I always recommend reviewing a PR yourself before submitting it for review by others.

**Pull request closed without merging**